apps/meteor/client/providers/DeviceProvider/DeviceProvider.tsx

```diff
@@ -21,6 +21,13 @@ export const DeviceProvider = ({ children }: DeviceProviderProps): ReactElement
 		type: 'audio',
 	});
 
+	const setAudioInputDevice = (device: Device): void => {
+		if (!isSecureContext) {
+			throw new Error('Device Changes are not available on insecure contexts');
+		}
+		setSelectedAudioInputDevice(device);
+	};
+
```

Comment:
I think we could handle this better this case errors, maybe setting some state 

also isSecureContext is not mutable so you could make the WHOLE provider behave differently if `false`, if it wasn't a safe environment, and not only guarantee that the functions won't break


btw at least do typeof isSecureContext && isSecureContext, we still have IE support